pytest_rerunfailures.py

```diff
@@ -255,6 +269,7 @@ def pytest_runtest_protocol(item, nextitem):
                 or not report.failed
                 or xfail
                 or is_terminal_error
+                or not condition
```

Comment:
This list is getting longer again, this seems like a good place for a refactoring or as an instance of feature creep.

Reply:
One solution is to simply move this out to a `should_rerun`-function. That way there's less cognitive load when understanding either function.

Reply:
:+1: